portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts

```diff
@@ -1,15 +1,23 @@
-const { format, startOfMonth, addMonths } = require('date-fns');
-const { getFormattedReportPeriodWithLongMonth } = require('@ukef/dtfs2-common');
-const { extractCsvData, removeCellAddressesFromArray } = require('../../../utils/csv-utils');
-const { getUploadErrors } = require('./utilisation-report-upload-errors');
-const { getDueReportPeriodsByBankId, getReportDueDate } = require('./utilisation-report-status');
-const api = require('../../../api');
-const { getReportAndUserDetails } = require('./utilisation-report-details');
-const { PRIMARY_NAV_KEY } = require('../../../constants');
-const { filterReportJsonToRelevantKeys } = require('../../../helpers/filterReportJsonToRelevantKeys');
-
-const setSessionUtilisationReport = (req, nextDueReportPeriod) => {
-  req.session.utilisationReport = {
+import { Request, Response } from 'express';
+import { format, startOfMonth, addMonths } from 'date-fns';
+import { getFormattedReportPeriodWithLongMonth, isFeeRecordCorrectionFeatureFlagEnabled, ReportPeriod } from '@ukef/dtfs2-common';
+import { extractCsvData, removeCellAddressesFromArray } from '../../../utils/csv-utils';
+import { getUploadErrors, UtilisationReportUploadErrors } from './utilisation-report-upload-errors';
+import { getDueReportPeriodsByBankId, getReportDueDate } from './utilisation-report-status';
+import api from '../../../api';
+import { getReportAndUserDetails, ReportAndUserDetails } from './utilisation-report-details';
+import { PRIMARY_NAV_KEY } from '../../../constants';
+import { filterReportJsonToRelevantKeys } from '../../../helpers/filterReportJsonToRelevantKeys';
+import { asLoggedInUserSession, LoggedInUserSession } from '../../../helpers/express-session';
+import { isNonEmptyPendingCorrectionsResponseBody, mapToPendingCorrectionsViewModel } from './pending-corrections-helper';
+
+/**
+ * Sets the utilisation report session data for the report upload journey
+ * @param req - The request object
+ * @param nextDueReportPeriod - The next due report period
+ */
+const setSessionUtilisationReport = (req: Request, nextDueReportPeriod: ReportPeriod & { formattedReportPeriod: string }) => {
+  (req.session as LoggedInUserSession).utilisationReport = {
```

Comment:
from sonarCloud - This assertion is unnecessary since it does not change the type of the expression. (as LoggedInUserSession not needed)

Reply:
If I remove it I get the following:
<img width="954" alt="Screenshot 2025-01-06 at 17 47 35" src="https://github.com/user-attachments/assets/6187fa52-463e-4f41-b889-29740ab4fafb" />

Reply:
can ignore this one then